src/Command/SystemLocaleCommand.php

```diff
@@ -0,0 +1,229 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Production\Command;
+
+use Doctrine\DBAL\Connection;
+use PDO;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Adapter\Console\ShopwareStyle;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class SystemLocaleCommand extends Command
+{
+    public static $defaultName = 'system:locale-destructive';
+
+    /**
+     * @var SymfonyStyle
+     */
+    protected $io;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    private $activated = false;
+
+    public function __construct(Connection $connection)
+    {
+        parent::__construct();
+        $this->connection = $connection;
+    }
+
+    public function activateCommand(): void
+    {
+        $this->activated = true;
+    }
+
+    protected function configure(): void
+    {
+        $this->addArgument('locale', InputArgument::REQUIRED, 'ISO locale for the shop');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $output = new ShopwareStyle($input, $output);
+        $output->section('Shop locale');
+
+        if (!$this->activated) {
+            $output->error('The command has not been activated by the runtime and therefore cannot be executed. It is intended to be used with system:install --locale');
+
+            return 1;
+        }
+
+        $locale = $input->getArgument('locale');
+        $this->setDefaultLanguage($locale);
+
+        $output->success(sprintf('Successfully changed shop default locale to %s', $locale));
+
+        return 0;
+    }
+
+    private function getLocaleId(string $iso): string
+    {
+        $stmt = $this->connection->prepare('SELECT locale.id FROM locale WHERE LOWER(locale.code) = LOWER(?)');
+        $stmt->execute([$iso]);
+        $id = $stmt->fetchColumn();
+
+        if (!$id) {
+            throw new \RuntimeException('Locale with iso-code ' . $iso . ' not found');
+        }
+
+        return (string)$id;
+    }
+
+    private function swapDefaultLanguageId(string $newLanguageId): void
+    {
+        $this->connection->exec('SET FOREIGN_KEY_CHECKS = 0');
```

Comment:
as mentioned in comments, in recovery's ShopService they are not disabled which is a better solution as CASCADE UPDATE takes care of all necessary changes and this way, the content of mail templates etc. is correctly assigned.

Even better would be to re-assign everything correctly afterwards but...